opts/hosts.go

```diff
@@ -77,6 +79,12 @@ func parseDockerDaemonHost(addr string) (string, error) {
 		return parseSimpleProtoAddr("npipe", addrParts[1], DefaultNamedPipe)
 	case "fd":
 		return addr, nil
+	case "ssh": // unix over ssh (can be specified for `docker -H` but not for `dockerd -H`)
+		usr, err := whoami()
```

Comment:
Why not let `ssh(1)` handle this? This calls into `/etc/passwd` parsing on `macOS` that doesn't work for regular users.